src/Data/Tree.purs

```diff
@@ -88,15 +88,18 @@ instance comonadTree :: Comonad Tree where
 
 instance arbitraryTree :: Arbitrary a => Arbitrary (Tree a) where
   arbitrary = sized go
-    where go 0 = Tree <$> arbitrary <*> return []
+    where go 0 = Tree <$> arbitrary <*> pure []
           go n = do
             sz <- Int.floor <<< Math.abs <$> arbitrary
             Tree <$> arbitrary <*> replicateM sz (go (n / (sz + 1)))
 
+          replicateM n m | n < 1     = pure []
+                         | otherwise = sequence $ replicate n m
```

Comment:
I think you should be able to just do `sequence (replicate sz (go (n / (sz + 1)))`.  The `replicate` function for arrays will already deal with `sz < 1`.  Alternatively there is `replicateA` from unfoldable.